File: 프로그래머스/2/150369. 택배 배달과 수거하기/택배 배달과 수거하기.java
class Solution {
    public long solution(int cap, int n, int[] deliveries, int[] pickups) {
        long answer = 0;
        
        int delSum = 0;
        int pickSum = 0;
        for (int i=n;i>0;i--) {
            
            delSum += deliveries[i-1];
            pickSum += pickups[i-1];
            
            
            while (delSum > 0 || pickSum > 0){
                delSum -= cap;
                pickSum -= cap;
                answer += i*2;
            }
            
        }
        
        return answer;
    }
}

File: 프로그래머스/2/150369. 택배 배달과 수거하기/택배 배달과 수거하기.py
def solution(cap, n, deliveries, pickups):
    answer = 0
    deliver = 0
    retrieve = 0
    
    for i in range(n-1,-1,-1):

        deliver += deliveries[i]
        retrieve += pickups[i]

        while deliver > 0 or retrieve > 0:

            answer += (i+1) * 2

            deliver -= cap
            retrieve -= cap     
            
    return answer